arcgis-ios-sdk-samples/Edit data/Add features with contingent values/ContingentValuesTableViewController.swift

```diff
@@ -0,0 +1,303 @@
+    var selectedProtection: AGSContingentCodedValue? {
+        didSet {
+            let codedValueName = selectedProtection?.codedValue.name
+            // Display the selected protection.
+            editRightDetail(cell: protectionCell, rightDetailText: codedValueName)
+            // Reset the cell states accordingly.
+            resetCellStates()
+        }
+    }
+    
+    /// The selected buffer size.
+    var selectedBufferSize: Int? {
+        didSet {
+            if let selectedBufferSize = selectedBufferSize {
+                // Set the the buffer size attribute to the selected integer.
+                feature?.attributes["BufferSize"] = selectedBufferSize
+                let bufferSize = String(selectedBufferSize)
+                // Display the selected buffer size.
+                editRightDetail(cell: bufferSizeCell, rightDetailText: bufferSize)
+                // Validate the contingency.
+                validateContingency()
+            } else {
+                // If the value is nil, clear the right detail text.
+                editRightDetail(cell: bufferSizeCell, rightDetailText: " ")
+                // Hide the buffer size picker.
+                toggleBufferSizePickerVisibility()
+            }
+        }
+    }
+    
+    // MARK: Functions
+    
+    /// Show the options for the status field.
+    func showStatusOptions() {
+        // Get the previously selected status value.
+        let previouslySelectedStatus = selectedStatus
+        // Get the first field by name.
+        let statusField = featureTable?.field(forName: "Status")
+        // Get the field's domains as coded value domain.
+        let codedValueDomain = statusField?.domain as! AGSCodedValueDomain
+        // Get the coded value domain's coded values.
+        let statusCodedValues = codedValueDomain.codedValues
+        // Get the selected index if applicable.
+        let selectedIndex = statusCodedValues.firstIndex { $0.name == selectedStatus?.name }
+        // Use the coded value names as labels to show the option table view controller.
+        let optionsViewController = OptionsTableViewController(labels: statusCodedValues.map { $0.name }, selectedIndex: selectedIndex) { [weak self] newIndex in
+            guard let self = self else { return }
+            // Set the selected status.
+            self.selectedStatus = statusCodedValues[newIndex]
+            // Set the attributes to nil if the status value has changed.
+            if self.selectedStatus != previouslySelectedStatus, self.selectedProtection != nil {
+                self.selectedProtection = nil
+                if self.selectedBufferSize != nil {
+                    self.selectedBufferSize = nil
+                }
+            }
+            self.createFeature(with: self.selectedStatus!)
+            // Remove the options view controller after selection.
+            self.navigationController?.popViewController(animated: true)
```

Comment:
Based on experience, I've found it helpful to be explicit as to which view controller needs to be popped to. That way you can be sure that the user will return to this one.
```suggestion
            self.navigationController?.popToViewController(self, animated: true)
```